File: openapi/code_samples/reactJs/api_company_find.js
import React, { useEffect, useState } from 'react';
import axios from 'axios';

function App() {
    const [data, setData] = useState([]);
    const queryParams = {
        pageNo: 0,
        pageSize: 10,
        cuiOrName: '',
        sortBy: 'name',
        currency: 'RON',
    };

    useEffect(() => {
        axios.get('/api/company/find', {
            params: queryParams,
        })
        .then(response => {
            setData(response.data);
        })
        .catch(error => {
            console.error(error);
        });
    }, []);

    return (
        <div className="App">
            {/* Your React component rendering here */}
        </div>
    );
}

export default App;

File: openapi/code_samples/java/api_company_find.java
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class CompanyController {

    @GetMapping("/api/company/find")
    public ResponseEntity<?> find(
            @RequestParam(name = "pageNo", defaultValue = "0") int pageNo,
            @RequestParam(name = "pageSize", defaultValue = "10") int pageSize,
            @RequestParam(name = "cuiOrName", defaultValue = "") String cuiOrName,
            @RequestParam(name = "sortBy", defaultValue = "name") String sortBy,
            @RequestParam(name = "currency", defaultValue = "RON") String currency) {

        // Your implementation here

        return ResponseEntity.ok().build();
    }
}
